servers/rendering/rendering_device.cpp

```diff
@@ -4998,8 +5003,16 @@ Error RenderingDevice::draw_list_switch_to_next_pass_split(uint32_t p_splits, Dr
 #endif
 
 Error RenderingDevice::_draw_list_allocate(const Rect2i &p_viewport, uint32_t p_subpass) {
-	draw_list = memnew(DrawList);
+
+	if (draw_list == nullptr) {
+		draw_list = memnew(DrawList);
+	}
+	
 	draw_list->viewport = p_viewport;
+	draw_list->viewport_set = true;
+#ifdef DEBUG_ENABLED
+	draw_list->validation.active = true;
```

Comment:
I'm not happy with this. It technically works, but I think this leaks memory because I am calling memset despite having some complex types inside the struct (only in debug builds). 

I think a possible better option is to:
1. Allocate the struct on startup
2. Use placement new to call the construction each time
3. Call the destructor when ending the draw list. 

Alternatively, we should consider just keeping the variable on the stack. I'm not sure if it actually ever needs to be on the heap now that there is only ever one used at a time

Reply:
The whole allocation of the draw list are remnants of a system where split draw lists were supported. I think we can very well just get rid of that and just keep one draw list that is a member of RenderingDevice and clearing its contents when beginning a new one.

Reply:
Sounds good. I started doing that last night anyway since it seemed like the right approach to me.